framework/include/loops/ComputeMortarFunctor.h

```diff
@@ -66,4 +70,8 @@ class ComputeMortarFunctor
 
   /// A reference to the assembly object
   Assembly & _assembly;
+
+  std::map<SubdomainID, std::deque<MaterialBase *>> _secondary_ip_sub_to_mats;
+  std::map<SubdomainID, std::deque<MaterialBase *>> _primary_ip_sub_to_mats;
+  std::deque<MaterialBase *> _secondary_boundary_mats;
```

Comment:
If I understand this correctly, for everything object relying on mortar (i.e. constraints, mortar aux kernel), we require that there is an object that holds a map with the material properties that we are going to need, computed via `buildRequiredMaterials`.